web/controllers/channel_controller.ex

```diff
@@ -1,23 +1,67 @@
 defmodule Ask.ChannelController do
   use Ask.Web, :api_controller
 
-  alias Ask.Channel
+  alias Ask.{Channel, Project, Logger}
+
+  def index(conn, %{"project_id" => project_id}) do
+    channels = conn
+    |> load_project(project_id)
+    |> assoc(:channels)
+    |> Repo.all
+
+    render(conn, "index.json", channels: channels |> Repo.preload(:projects))
+  end
 
   def index(conn, _params) do
     channels = conn
     |> current_user
     |> assoc(:channels)
     |> Repo.all
 
-    render(conn, "index.json", channels: channels)
+    render(conn, "index.json", channels: channels |> Repo.preload(:projects))
   end
 
   def show(conn, %{"id" => id}) do
     channel = Channel
     |> Repo.get!(id)
     |> authorize_channel(conn)
 
-    render(conn, "show.json", channel: channel)
+    render(conn, "show.json", channel: channel |> Repo.preload(:projects))
+  end
+
+  def update(conn, %{"id" => id, "channel" => channel_params}) do
+    channel =
+      Channel
+      |> Repo.get!(id)
+      |> authorize_channel(conn)
+      |> Repo.preload([:projects])
+
+    changeset =
+      channel
+      |> Channel.changeset(channel_params)
+      |> update_projects(channel_params)
```

Comment:
Is this missing an authorization over the project here? A user should be able to share a channel only with the projects where he/she is a member.

Reply:
We should allow it if the user has a reader privilege? or we should enforce it to be admin/editor to be able to share?